File: app/src/main/java/org/shirakawatyu/yamibo/novel/bean/ReaderSettings.java
package org.shirakawatyu.yamibo.novel.bean;

public class ReaderSettings {
    private Float fontSizePx;
    private Float lineHeightPx;
    private Float paddingDp;

    private Boolean nightMode;
    private Boolean loadImages;
    private String backgroundColor;
    private Boolean isVerticalMode;
    private Integer translationMode;

    public ReaderSettings(Float fontSizePx, Float lineHeightPx, Float padding, Boolean nightMode, String backgroundColor, Boolean loadImages, Boolean isVerticalMode, Integer translationMode) {
        this.fontSizePx = fontSizePx;
        this.lineHeightPx = lineHeightPx;
        this.paddingDp = padding;
        this.nightMode = nightMode;
        this.backgroundColor = backgroundColor;
        this.loadImages = loadImages;
        this.isVerticalMode = isVerticalMode;
        this.translationMode = translationMode;
    }

    public ReaderSettings() {
    }

    public Float getFontSizePx() {
        return fontSizePx;
    }

    public void setFontSizePx(Float fontSizePx) {
        this.fontSizePx = fontSizePx;
    }

    public Float getLineHeightPx() {
        return lineHeightPx;
    }

    public void setLineHeightPx(Float lineHeightPx) {
        this.lineHeightPx = lineHeightPx;
    }

    public Float getPaddingDp() {
        return paddingDp;
    }

    public void setPaddingDp(Float paddingDp) {
        this.paddingDp = paddingDp;
    }

    public Boolean getNightMode() {
        return nightMode;
    }

    public void setNightMode(Boolean nightMode) {
        this.nightMode = nightMode;
    }

    public String getBackgroundColor() {
        return backgroundColor;
    }

    public void setBackgroundColor(String backgroundColor) {
        this.backgroundColor = backgroundColor;
    }

    public Boolean getLoadImages() {
        return loadImages;
    }

    public void setLoadImages(Boolean loadImages) {
        this.loadImages = loadImages;
    }

    public Boolean getIsVerticalMode() {
        return isVerticalMode;
    }

    public void setIsVerticalMode(Boolean isVerticalMode) {
        this.isVerticalMode = isVerticalMode;
    }

    public Integer getTranslationMode() {
        return translationMode == null ? 0 : translationMode;
    }

    public void setTranslationMode(Integer translationMode) {
        this.translationMode = translationMode;
    }
}

File: app/src/main/java/org/shirakawatyu/yamibo/novel/bean/ReaderSettings.kt
package org.shirakawatyu.yamibo.novel.bean

import com.alibaba.fastjson2.annotation.JSONCreator
import com.alibaba.fastjson2.annotation.JSONField

data class ReaderSettings @JSONCreator constructor(
    @JSONField(name = "fontSizePx")
    var fontSizePx: Float? = null,

    @JSONField(name = "lineHeightPx")
    var lineHeightPx: Float? = null,

    @JSONField(name = "paddingDp")
    var paddingDp: Float? = null,

    @JSONField(name = "nightMode")
    var nightMode: Boolean? = null,

    @JSONField(name = "backgroundColor")
    var backgroundColor: String? = null,

    @JSONField(name = "loadImages")
    var loadImages: Boolean? = null,

    @JSONField(name = "isVerticalMode", alternateNames = ["verticalMode"])
    var isVerticalMode: Boolean? = null,

    @JSONField(name = "translationMode")
    var translationMode: Int? = null,

    @JSONField(name = "fontFamily")
    var fontFamily: Int? = null
)
